// react-native-table-component/components/cell.js
import React, { Component } from 'react';
import { View, ViewPropTypes, Text, StyleSheet } from 'react-native';

export class Cell extends Component {
  static propTypes = {
    style: ViewPropTypes.style,
    textStyle: Text.propTypes.style,
    borderStyle: ViewPropTypes.style
  };

  render() {
    const { data, width, height, flex, style, textStyle, borderStyle, ...props } = this.props;

    const textDom = React.isValidElement(data) ? (
      data
    ) : (
      <Text style={[textStyle, styles.text]} {...props}>
        {data[1]}
      </Text>
    );
    const borderTopWidth = (borderStyle && borderStyle.borderWidth) || 1;
    const borderRightWidth = borderTopWidth;
    const borderColor = (borderStyle && borderStyle.borderColor) || '#000';
    return (
      <View
        style={[
          {
            borderTopWidth,
            borderRightWidth,
            borderColor
          },
          styles.cell,
          width && { width },
          height && { height },
          flex && { flex },
          !width && !flex && !height && !style && { flex: 1 },
          style,
          data[0] === 'A' ? styles.aisle : null,
          data[0] === 'W' ? styles.window : null,
          data[0] === 'M' ? styles.middle : null
        ]}
      >
        {textDom}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  cell: { width: 35, height: 35 },
  text: { backgroundColor: 'transparent', color: 'white', textAlign: 'center' },
  aisle: { backgroundColor: 'steelblue' },
  window: { backgroundColor: 'yellowgreen' },
  middle: { backgroundColor: 'indianred' }
});
